core/main/src/firebolt/handlers/provider_registrar.rs

```diff
@@ -0,0 +1,228 @@
+
+                                    ProviderBroker::register_or_unregister_provider(
+                                        &context.platform_state,
+                                        attributes.capability.into(),
+                                        attributes.method.into(),
+                                        attributes.event,
+                                        call_context,
+                                        request,
+                                    )
+                                    .await;
+
+                                    Ok(ListenerResponse {
+                                        listening,
+                                        event: attributes.event.into(),
+                                    })
+                                },
+                            )
+                            .unwrap();
+                    }
+
+                    // Register focus function
+                    if let Some(method) = provider_set.focus.clone() {
+                        let focus_method =
+                            FireboltOpenRpcMethod::name_with_lowercase_module(&method.name).leak();
+
+                        rpc_module
+                            .register_async_method(
+                                focus_method,
+                                move |params, context| async move {
+                                    let mut params_sequence = params.sequence();
+                                    let call_context: CallContext = params_sequence.next().unwrap();
+                                    let request: FocusRequest = params_sequence.next().unwrap();
+
+                                    ProviderBroker::focus(
+                                        &context.platform_state,
+                                        call_context,
+                                        attributes.capability.into(),
+                                        request,
+                                    )
+                                    .await;
+                                    Ok(None) as RpcResult<Option<()>>
+                                },
+                            )
+                            .unwrap();
+                    }
+
+                    // Register response function
+                    if let Some(method) = provider_set.response.clone() {
+                        let response_method =
+                            FireboltOpenRpcMethod::name_with_lowercase_module(&method.name).leak();
+
+                        rpc_module
+                            .register_async_method(
+                                response_method,
+                                move |params, context| async move {
+                                    let params_sequence = params.sequence();
+
+                                    if let Some(provider_response) =
+                                        ProviderRegistrar::get_provider_response(
```

Comment:
Guessing we need this only for Pin and Acknowledge challenge because Ripple needs to interpret them as well along with Apps